akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -3072,6 +3124,20 @@ trait FlowOps[+Out, +Mat] {
         }
     }
 
+  private def internalConcatAll[U >: Out](those: immutable.Seq[Graph[SourceShape[U], _]], detached: Boolean): Repr[U] =
```

Comment:
Since this is internal I think we can keep it `Array[...]` and avoid that extra collection transformation.